src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -145,6 +130,35 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
                     }
                     process = ProcessLauncher.Launcher.ChildProc;
                 }
+                else if (port.Length != 0)
+                {
+                    ReversedDiagnosticsServer server = new ReversedDiagnosticsServer(port);
+                    server.Start();
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    {
+                        Console.WriteLine($"Waiting for connection on {port}");
+                        Console.WriteLine($"Start an application with the following environment variable: DOTNET_DiagnosticPorts={port},suspend");
+                    }
+                    else
+                    {
+                        Console.WriteLine($"Waiting for connection on {Path.GetFullPath(port)}");
+                        Console.WriteLine($"Start an application with the following environment variable: DOTNET_DiagnosticPorts={Path.GetFullPath(port)},suspend");
+                    }
+
+                    ManualResetEvent acceptEvent = new ManualResetEvent(false);
+                    IpcEndpointInfo endpointInfo = new IpcEndpointInfo();
+                    ct.Register(() => acceptEvent.Set());
+                    Task acceptTask = Task.Run(() =>
+                    {
+                        endpointInfo = server.Accept(TimeSpan.FromMilliseconds(Int32.MaxValue));
+                        Console.WriteLine($"Connection established");
+                        acceptEvent.Set();
+                    });
+                    acceptEvent.WaitOne();
+                    diagnosticsClient = new DiagnosticsClient(endpointInfo.Endpoint);
+                    diagnosticsClient.ResumeRuntime();
+                    process = Process.GetProcessById(endpointInfo.ProcessId);
+                }
```

Comment:
This is identical to some of the other code above.  Could we consolidate it into a helper function?